src/zh/bilimanga/src/eu/kanade/tachiyomi/extension/zh/bilimanga/BiliManga.kt

```diff
@@ -0,0 +1,164 @@
+
+    // Search Page
+
+    override fun getFilterList() = buildFilterList()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = baseUrl.toHttpUrl().newBuilder()
+        if (query.isNotBlank()) {
+            url.addPathSegment("search").addPathSegment("${query}_$page.html")
+        } else {
+            url.addPathSegment("top").addPathSegment(filters[1].toString())
+                .addPathSegment("$page.html")
+        }
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.pathSegments.contains("detail")) {
+            return MangasPage(listOf(mangaDetailsParse(response)), false)
+        }
+        return popularMangaParse(response)
+    }
+
+    // Manga Detail Page
+
+    override fun mangaDetailsParse(response: Response) = SManga.create().apply {
+        val doc = response.asJsoup()
+        val meta = doc.selectFirst(".book-meta")!!.text().split("|")
+        val extra = meta.filterNot(META_REGEX::containsMatchIn)
+        val backup = preferences.getBoolean(BACKUP_TITLE, false)
+        url = doc.location()
+        title = doc.selectFirst(if (backup) ".backupname" else ".book-title")!!.text()
+        thumbnail_url = doc.selectFirst(".book-cover")!!.attr("src")
+        description = doc.selectFirst("#bookSummary")?.text()
+        artist = doc.selectFirst(".authorname")?.text()
+        author = doc.selectFirst(".illname")?.text() ?: artist
+        status = when (meta.firstOrNull()) {
+            "連載" -> SManga.ONGOING
+            "完結" -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+        genre = (doc.select(".tag-small").map(Element::text) + extra).joinToString()
+        initialized = true
+    }
+
+    // Catalog Page
+
+    override fun chapterListRequest(manga: SManga) = GET("$baseUrl/read/${manga.id}/catalog", headers)
+
+    override fun chapterListParse(response: Response) = response.asJsoup().let {
+        val info = it.selectFirst(".chapter-sub-title")!!.text()
+        val date = DATE_FORMAT.tryParse(DATE_REGEX.find(info)?.value)
+        val elements = it.select(".chapter-li-a")
+        elements.mapIndexed { i, e ->
+            val url = e.absUrl("href").takeUnless("javascript:cid(1)"::equals)
+            SChapter.create().apply {
+                name = e.text().convert()
+                date_upload = date
+                if (url == null) scanlator = "鏈接缺失，由插件預測生成"
+                setUrlWithoutDomain(url ?: getChapterUrlByContext(i, elements))
+            }
+        }.reversed()
+    }
+
+    // Manga View Page
+
+    override fun pageListParse(response: Response) = response.asJsoup().let {
+        val images = it.select(".imagecontent")
+        val prompt = it.selectFirst("#acontentz")
+        require(images.size > 0) { prompt?.let { "漫畫可能已下架或需要登錄查看" } ?: "章节鏈接错误" }
```

Comment:
```suggestion
        require(images.size > 0) { if (it.selectFirst("#acontentz") != null) "漫畫可能已下架或需要登錄查看" else "章节鏈接错误" }
```

Reply:
Too long code needs to be folded for display, I think it's more readable to split it into two lines

Reply:
```suggestion
        if (images.size == 0) {
            val prompt = it.selectFirst("#acontentz")
            throw Exception(if (prompt != null) "漫畫可能已下架或需要登錄查看" else "章节鏈接错误")
        }
```

1. `?.let {} ?:` should be replaced with `if else` for clarity in this case.
2. `selectFirst()` should be moved inside the block for lazy evaluation.
3. [`require()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/require.html) throws `IllegalArgumentException`. You actually mean [`check()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/check.html) which throws `IllegalStateException`.
4. In this case, it's better to throw a plain `Exception` so the app will only display the message. If you throw an `IllegalStateException`, the app will prefix it with the exception name, which is not useful.

Reply:
Alright, you're right, you think very meticulously, I indeed didn't consider other situations